src/pages/en/guides/upgrade-to/v2.mdx

```diff
@@ -0,0 +1,359 @@
+
+```js del={6}
+// astro.config.mjs
+import { defineConfig } from 'astro/config';
+
+export default defineConfig({
+  markdown: {
+    extendDefaultPlugins: true,
+  }
+});
+```
+
+#### Added: `gfm` and `smartypants`
+
+Astro 2.0 replaces `markdown.extendDefaultPlugins` with separate Boolean options to control each of Astro's built-in default Markdown plugins individually. These are enabled by default, and can be set to `false` to remove each plugin independently.
+
+##### What should I do?
+
+Replace `extendDefaultPlugins: false` and add the flags to disable each plugin individually instead.
+
+- `markdown.gfm: false` disables GitHub-Flavored Markdown
+- `markdown.smartypants: false` disables SmartyPants
+
+```js del={6} ins={7-8}
+// astro.config.mjs
+import { defineConfig } from 'astro/config';
+
+export default defineConfig({
+  markdown: {
+    extendDefaultPlugins: false,
+    smartypants: false,
+    gfm: false,
+  }
+});
+```
+
+### Changed: MDX plugin configuration
+
+#### Replaced: `extendPlugins` -> `extendMarkdownConfig`
+
+In 1.0, `.mdx.extendPlugins` managed how your MDX files should inherit your Markdown plugin configuration. This setting was used to indicate whether all Markdown plugins (`markdown`), or only Astro's default plugins (`defaults`) should be applied to MDX.
+
+Astro 2.0 replaces `mdx.extendPlugins` with three new, independently-configurable options:
+
+- **[`mdx.extendMarkdownConfig`](/en/guides/integrations-guide/mdx/#extendmarkdownconfig)** (`true` by default) to inherit all of your Markdown configuration or none of it.
+- **`mdx.gfm`** (`true` by default) to enable or disable GitHub-Flavored Markdown in MDX.
+- **`mdx.smartypants`** (`true` by default) to enable or disable SmartyPants in MDX.
+
+ ##### What should I do?
+
+Delete `extendPlugins: 'markdown'` in your configuration. This is now default behavior. 
+
+```js del={6}
+// astro.config.mjs
+import { defineConfig } from 'astro/config';
+
+export default defineConfig({
+  mdx: {
+    extendPlugins: markdown,
+  }
+});
+```
+
+Replace `extendPlugins: 'defaults'` with `extendMarkdownConfig: false` and add the [separate options for GitHub-Flavored Markdown and SmartyPants](/en/guides/integrations-guide/mdx/#options-inherited-from-markdown-config) to enable these default plugins individually in MDX.
+
+```js del={6} ins={7-9}
+// astro.config.mjs
+import { defineConfig } from 'astro/config';
+
+export default defineConfig({
+  mdx: {
+    extendPlugins: defaults,
+    extendMarkdownConfig: false,
+    smartypants: true,
+    gfm: true,
+  }
+});
+```
+
+#### Added: More MDX config options to match Markdown
+
+Setting `extendMarkdownConfig: false` now allows you to individually set [every available Markdown configuration option](/en/reference/configuration-reference/#markdown-options) (except `drafts`) separately in your MDX integration configuration.
```

Comment:
This section is confusing. The text says you can set `extendMarkdownConfig: false` but the code sample doesn’t show where and I don’t really understand the combo of inherited/ignored/applied/overridden in the comments.